test/e2e/istio_test.go

```diff
@@ -364,6 +378,27 @@ func getKialiWorkloadHealth(t *testing.T, kialiAPIUrl string, namespace, workloa
 	return &health, nil
 }
 
+// getKialiWorkloadHealthIstioByWorkloadEndpoint gets metrics of workload by /namespaces/<ns>/workloads/<workload> API.
+// Used in Istio 1.18 and later. Istio 1.17 does not have this API.
+func getKialiWorkloadHealthIstioByWorkloadEndpoint(t *testing.T, kialiAPIUrl string, namespace, workloadName string) (*workloadHealth, error) {
+	kialiWorkloadURL := fmt.Sprintf("%s/namespaces/%s/workloads/%s", kialiAPIUrl, namespace, workloadName)
+	resp, err := helpers.DefaultHTTPClient().Get(kialiWorkloadURL)
+	require.NoErrorf(t, err, "failed to call Kiali workload API %s", kialiWorkloadURL)
+	defer resp.Body.Close()
+	// verify the workload response
+	require.Equalf(t, http.StatusOK, resp.StatusCode, "Got code %d from Kiali workload API %s", resp.StatusCode, kialiWorkloadURL)
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
checking error here and returning vs `require.NoError(t, err)` above. Why the disparity?

Reply:
Changed to `require` in reading responses.